### DISCUSSION TITLE
How to extend `timeout` delay with `Bun`: is `Bun.serve` to right way?

### DISCUSSION THREAD

**Discussion by charnould** [honojs/hono#3722] [Q&A]
Hello,
I'm using `Bun` + `Hono` to build a chatbot (with streams).
Sometimes it fails with `[Bun.serve]: request timed out after 10 seconds. Pass idleTimeout to configure.` because my request is sadly too long.

I've tried to extend timeout delay and dunno how to make it effective.  
Actually, I wrote this in my `app.ts` :

```
import type { Serve } from 'bun'
import { Hono } from 'hono'

....

export default {
  idleTimeout: 90, // seconds
  fetch: app.fetch
} satisfies Serve
```

Source: https://bun.sh/docs/api/http#idletimeout

It seems to work, but I don't know if it's the official/cleanest way (there is an official middleware but it warns for Stream : https://hono.dev/docs/middleware/builtin/timeout#notes)...

Any advice?   
Thanks

**Comment by charnould** [comment]
Hello, 
Any idea on this?
Thanks.

**Comment by EdamAme-x** [comment]
wait a moment

**Comment by EdamAme-x** [comment]
Hi @charnould 
Please check this.
https://github.com/honojs/hono/issues/3637#issuecomment-2459833685
